kroxylicious-filter-test-support/src/main/java/io/kroxylicious/test/record/RecordTestUtils.java

```diff
@@ -0,0 +1,108 @@
+                                Header... headers) {
+        return record(RecordBatch.CURRENT_MAGIC_VALUE, 0, 0, null, value, headers);
+    }
+
+    public static Record record(String value,
+                                Header... headers) {
+        return record(RecordBatch.CURRENT_MAGIC_VALUE,
+                0,
+                0,
+                null,
+                ByteBuffer.wrap(value.getBytes(StandardCharsets.UTF_8)),
+                headers);
+    }
+
+    public static Record record(String key,
+                                String value,
+                                Header... headers) {
+        return record(RecordBatch.CURRENT_MAGIC_VALUE,
+                0,
+                0,
+                ByteBuffer.wrap(key.getBytes(StandardCharsets.UTF_8)),
+                ByteBuffer.wrap(value.getBytes(StandardCharsets.UTF_8)),
+                headers);
+    }
+
+    public static Record record(ByteBuffer key,
+                                ByteBuffer value,
+                                Header... headers) {
+        return record(RecordBatch.CURRENT_MAGIC_VALUE, 0, 0, key, value, headers);
+    }
+
+    public static Record record(byte magic,
+                                long offset,
+                                long timestamp,
+                                ByteBuffer key,
+                                ByteBuffer value,
+                                Header... headers) {
+        // This is a bit of a rigmarole, but it ensures that calls to getSizeInBytes()
+        // on the returned Record is actually correct
+        MemoryRecords mr = memoryRecords(magic, offset, timestamp, key, value, headers);
+        return MemoryRecords.readableRecords(mr.buffer()).records().iterator().next();
+    }
+
+    public static RecordBatch recordBatch(String key,
+                                          String value,
+                                          Header... headers) {
+        return memoryRecords(RecordBatch.CURRENT_MAGIC_VALUE,
+                0,
+                0,
+                ByteBuffer.wrap(key.getBytes(StandardCharsets.UTF_8)),
+                ByteBuffer.wrap(value.getBytes(StandardCharsets.UTF_8)),
+                headers)
+                .firstBatch();
+    }
+
+    public static MemoryRecords memoryRecords(String key, String value, Header... headers) {
+        return memoryRecords(RecordBatch.CURRENT_MAGIC_VALUE,
+                0,
+                0,
+                ByteBuffer.wrap(key.getBytes(StandardCharsets.UTF_8)),
+                ByteBuffer.wrap(value.getBytes(StandardCharsets.UTF_8)),
+                headers);
+    }
+
+    public static MemoryRecords memoryRecords(byte magic, long offset, long timestamp, ByteBuffer key, ByteBuffer value, Header... headers) {
+        MemoryRecordsBuilder memoryRecordsBuilder = new MemoryRecordsBuilder(
+                ByteBuffer.allocate(1024),
+                magic,
+                CompressionType.NONE,
```

Comment:
Could be worth adding another method with all the batch configs exposed?

Reply:
Easily done in a PR where such a method is needed.